lib/scope.js

```diff
@@ -1125,28 +1119,34 @@ module.exports = {
     // TODO: Maybe we should return something else or something in addition about used vars
   },  // Ends scope.processVar()
 
-  getThereIsAnotherValue: async function ( scope, { value, from_table=true }) {
+  getThereIsAnotherValue: async function ( scope, { row, from_table=true }) {
     /* Given story row-like data for a `.there_is_another` var, as well as
     * whether this is from a table row, return the best guess as to a
     * safe value to set for this round. Give the developer a warning if
     * it's an inappropriate value. Avoid mutation.
     *
     * WARNING: Remember to decrement the actual value later.
     */
-    if ( parseInt( value ) < 1 || value === `False` || value === `false` ) {  // Loop will end
+    let value = row.value;
+    let int_val = parseInt( value );
+
+    if ( row.times_used >= int_val ) {  // Loop will end
```

Comment:
In my head, this condition would fit better if it came after the `isNaN` condition. That way the top two conditions (value is a boolean, most likely from non-table test) form one group, and the bottom three (value is an integer, most likely from table) form one group.